pageserver/src/tenant/timeline.rs

```diff
@@ -5918,9 +5997,10 @@ impl<'a> TimelineWriter<'a> {
         // self.write_guard will be taken by the freezing
         self.tl
             .freeze_inmem_layer_at(freeze_at, &mut self.write_guard)
-            .await;
+            .await
+            .context("roll layer after batch")?;
```

Comment:
We can avoid the anyhow here

Reply:
Avoided in [3b3b54f](https://github.com/neondatabase/neon/pull/8229/commits/3b3b54fcc106e2ba50cbe1a64e78e09ce26e1472)